setup.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+
+maybe_install() {
+
+    exists=$(which "$1")
+    if [ "xx$exists" == "xx" ]; then
```

Comment:
I've previously used code like the following (I didn't actually run this so it might not work) to do this sort of thing.

```sh
if [ -x "$(which $1)" ]; then
    ... # installed case
else
    ... # not installed case
end
```

That said, I love how readable it is the way this is written!

Reply:
Yeah. This is "old habits die hard" combined with "shellcheck made demands"